react/modules/tasks/class-components.md

```diff
@@ -6,19 +6,97 @@
 2. Language Requirement
    - Use **TypeScript** for the project.
 3. Project Setup
-   - Initialize the project using [Vite](https://vitejs.dev/guide/) with the [_react-ts_ template](https://vite.new/react-ts).
+
+   - Initialize the project using [Vite](https://vitejs.dev/guide/) with the [_react-ts_ template](https://github.com/vitejs/vite/tree/main/packages/create-vite/template-react-ts).
+     ```sh
+     # npm 7+, extra double-dash is needed:
+     npm create vite@latest rs-react-app -- --template react-ts
+     ```
+
 4. Code Quality Tools
+
    1. ESLint
-      - Set up ESLint to throw errors if TypeScript's _any_ type is used.
-      - Set up [eslint-plugin-react-compiler](https://www.npmjs.com/package/eslint-plugin-react-compiler) to throw errors if React rules are violated.
+      - ESlint v.9 is already a part of the _react-ts_ setup.
    2. Prettier
+
       - Integrate Prettier for code formatting.
+        You can execute the following command to add missing plugins:
+
+      ```sh
+      npm install -D eslint-plugin-react eslint-plugin-prettier eslint-config-prettier eslint-plugin-react-compiler@beta
+      npm install -D --save-exact prettier
+      ```
+
+      Now, add a new file `.prettierrc` to the root of the project:
+
+      ```json
+      {
+        "trailingComma": "es5",
+        "tabWidth": 2,
+        "semi": true,
+        "singleQuote": true
+      }
+      ```
+
+      And this is how your `eslint.config.js` should look like:
+
+      ```js
+      import js from "@eslint/js";
+      import globals from "globals";
+      import reactHooks from "eslint-plugin-react-hooks";
+      import reactRefresh from "eslint-plugin-react-refresh";
+      import react from "eslint-plugin-react";
+      import tseslint from "typescript-eslint";
+      import eslintPluginPrettier from "eslint-plugin-prettier/recommended";
+      import reactCompiler from "eslint-plugin-react-compiler";
+
+      export default tseslint.config(
+        { ignores: ["dist"] },
+        {
+          extends: [
+            js.configs.recommended,
+            ...tseslint.configs.strict,
+            eslintPluginPrettier,
+          ],
+          files: ["**/*.{ts,tsx}"],
+          languageOptions: {
+            ecmaVersion: 2020,
+            globals: globals.browser,
+          },
+          plugins: {
+            react,
+            "react-hooks": reactHooks,
+            "react-refresh": reactRefresh,
+            "react-compiler": reactCompiler,
+          },
+          rules: {
+            ...reactHooks.configs.recommended.rules,
+            "react-refresh/only-export-components": [
+              "warn",
+              { allowConstantExport: true },
+            ],
+            "react-compiler/react-compiler": "error",
+            ...react.configs.recommended.rules,
+            ...react.configs["jsx-runtime"].rules,
+          },
+        },
+      );
```

Comment:
We should add the following to the config:
```
    settings: {
      react: {
        version: 'detect',
      },
    },
```
Because currently i'm seeing a warning
![image](https://github.com/user-attachments/assets/5d052c3a-055c-462b-9ccc-5a092e652754)

Reply:
Agree, added.